### ISSUE TITLE
Running the Algol compiler?

### ISSUE THREAD

**Issue by louis-chretien** [pkoning2/algol#2]
This morning, i took the chance to load your algol files on my RSTS/E SIMH emulator.

I then looked at the bldalg.com file, and the way it is setup, makes it very difficult to use it: it uses some strange directories in the [170,*] range, and i don't have those files on my setup.

So i managed to do it by hand: i compiled the .MAC files, linked them and ran SILUS to create the RTS file, and installed it.

I the did a SET FILE on algol.alc, to give it protection 124 and runtime algol.

But when i tried to run it, that's where it got interesting... ;)

I first got an error on file WF:SWAP07.TMP. So I assigned WF: to SY: and tried again.

I then got:

$ run algol.alc
T - ?End of file on device

So I'm a bit stumped at the moment.

A few question:

1) How do we run the compiler? do you have some documentation?

2) Would there be a way inside the RTS to make a CCL command, and invoke the compiler that way?

3) I thought of creating a small "Hello World!" file in Algol, to help future installers to test the system

**Comment by louis-chretien** [comment]
Sorry, i just saw your last comment on issue #1, and you seem to have the same errors. I'll wait for your cue then...

**Comment by pkoning2** [comment]
The 170 series directories are the ones in the RSTS source kit (they are the ones used on the RSTS engineering development system).  I need a way to refer to source kit files in a more friendly way, given that it's probably best if I don't include them in this repository.
The WF thing I didn't remember, the workaround is as you said but it would be better if the code did that automatically
The EOF thing I don't understand yet.  I will need to find out what it is accessing when it gets that error.  I don't think the ALC file is bad; I also tried a compiled file from the RT11 kit that I dug up somewhere and while the RT11 version runs it doesn't load in the RTS.
CCL: probably, I don't remember if I did that.  I haven't looked at this code in about 40 years... :-)

**Comment by louis-chretien** [comment]
Yes, i saw a reference to the [170,*] accounts in Appendix B of the Internals manual. But it seems only the common.mac file is required, so i used the one in [1,2].

I also read with some delight chapter 21, about the SDA tool. This might come in handy for a project later on... I think you had a hand in this, didn't you? ;)

BTW, did u get my email about the mkDOSsimtap tool? I sent you the first draft. Spam?

**Comment by pkoning2** [comment]
> On Sep 5, 2022, at 9:06 AM, Louis Chretien ***@***.***> wrote:
> 
> 
> Yes, i saw a reference to the [170,*] accounts in Appendix B of the Internals manual. But it seems only the common.mac file is required, so i used the one in [1,2].
> 
> I also read with some delight chapter 21, about the SDA tool. This might come in handy for a project later on... I think you had a hand in this, didn't you? ;)

Yes, I learned about FORTH at some point and found it interesting.  So I took the FIG FORTH implementation and made an RTS out of it, then created some tools like ODT.FTH.

Then I got ambitious.  Looking at a crash dump output from ANALYS -- 100 or more pages of line printer output full of octal memory data -- is quite a pain.  I remembered SDA from my VMS training course, and said to myself "why not make an SDA for RSTS"?  And I realized that being able to "run code" interactively, sort of like what TECO lets you do, would be handy.

The tricky part is that SDA really needs 32 bit arithmetic and having to remember the "double length" operators all the time was annoying.  So I began by creating an entire "native 32 bit" FORTH world, and given that I then defined the actual SDA features.

A while later, Kevin Herbert added an X11 client to it, so he could display the SDA window on his X workstation.  I haven't tried that myself.

> BTW, did u get my email about the mkDOSsimtap tool? I sent you the first draft. Spam?

I did receive it, sorry, haven't had a change to try it.

	paul

**Comment by louis-chretien** [comment]
Are the ODT and SDA files, in the 10.1 distro kit, installed in UNSUPP$: the latest? or are there patches to those files?

**Comment by pkoning2** [comment]
Those are the latest.  The binaries (*.4th) should be up to date as well.  ODT hasn't changed in a long time.  It's like the B+ ODT, except that it can do 22 bit addressing on memory references, read and write.

**Comment by Terri-Kennedy** [comment]
Did anyone get a chance to look a this?

I'm getting the same "T - ?End of file on device" as the original submitter did. I thought it might be because Algol wanted to be a CCL, so I did a "ccl alg-ol=sy:[0,202]algol.alc". But then saying "algol test.alg"gives the same error.
